### PR TITLE
Remove `ToWithdrawalTx` helper

### LINKED ISSUE #209: Remove ToWithdrawalTx from testapp helpers
[`ToWithdrawalTx`](https://github.com/polymerdao/monomer/blob/main/testapp/helpers.go#L66) is used to create txs with `MsgInitiateWithdrawal` for use in testing with a default gas limit and data field. Now that gas limit is also passed in a parameter to support the withdrawals e2e tests, it was [suggested](https://github.com/polymerdao/monomer/pull/206#discussion_r1755494881) that the `ToWithdrawalTx` helper should be removed to minimize our API surface.

All uses of `ToWithdrawalTx` should be updated with the `MsgInitiateWithdrawal` inlined.

### PR DESCRIPTION
**Pull request by AnkushinDaniil**
This pull request removes the `ToWithdrawalTx` helper function from the `testapp` package.
Fixes #209 

## Summary by CodeRabbit

- **New Features**
	- Enhanced transaction creation for withdrawal requests with a more structured message format.

- **Bug Fixes**
	- Improved clarity and maintainability of transaction broadcasting logic.

- **Refactor**
	- Updated function visibility and removed unused functions to streamline the codebase.

### CODE REVIEW

**Comment by AnkushinDaniil** [comment]
@natebeauregard

File: builder/builder_test.go [hunk 1]
```diff
@@ -315,13 +315,14 @@ func TestBuildRollupTxs(t *testing.T) {
...
-	withdrawalTx := testapp.ToWithdrawalTx(
-		t,
-		cosmAddr.String(),
-		common.HexToAddress("0x12345abcde").String(),
-		math.NewIntFromBigInt(depositTxETH.Value()),
-		big.NewInt(100_000),
-	)
+	withdrawalTx := testapp.ToTx(t,
+		&types.MsgInitiateWithdrawal{
+			Sender:   cosmAddr.String(),
```
Comment (natebeauregard) [inline builder/builder_test.go:320]: formatting nit:
```suggestion
	withdrawalTx := testapp.ToTx(t, &types.MsgInitiateWithdrawal{
		Sender:   cosmAddr.String(),
```

File: e2e/stack_test.go [hunk 2]
```diff
@@ -324,13 +324,14 @@ func rollupFlow(t *testing.T, stack *e2e.StackConfig) {
...
-		testapp.ToWithdrawalTx(
-			t,
-			utils.EvmToCosmosAddress(*withdrawalTx.Sender).String(),
-			withdrawalTx.Target.String(),
-			math.NewIntFromBigInt(withdrawalTx.Value),
-			withdrawalTx.GasLimit,
-		),
+		testapp.ToTx(t,
+			&rolluptypes.MsgInitiateWithdrawal{
+				Sender:   utils.EvmToCosmosAddress(*withdrawalTx.Sender).String(),
```
Comment (natebeauregard) [inline e2e/stack_test.go:329]: formatting nit:
```suggestion
		testapp.ToTx(t, &rolluptypes.MsgInitiateWithdrawal{
			Sender:   utils.EvmToCosmosAddress(*withdrawalTx.Sender).String(),
```

**Review by natebeauregard — COMMENTED** [review]
Looks good, thanks for making this. Left a few formatting nits but I'll approve/merge once they're updated.

File: builder/builder_test.go [hunk 3]
```diff
@@ -315,13 +315,13 @@ func TestBuildRollupTxs(t *testing.T) {
...
-		math.NewIntFromBigInt(depositTxETH.Value()),
-		big.NewInt(100_000),
-	)
+	withdrawalTx := testapp.ToTx(t, &types.MsgInitiateWithdrawal{
+		Sender:   cosmAddr.String(),
+			Target:   common.HexToAddress("0x12345abcde").String(),
+			Value:    math.NewIntFromBigInt(depositTxETH.Value()),
+			GasLimit: big.NewInt(100_000).Bytes(),
+			Data:     []byte{},
+		})
```
Comment (natebeauregard) [inline builder/builder_test.go:324]: nit:
```suggestion
	withdrawalTx := testapp.ToTx(t, &types.MsgInitiateWithdrawal{
		Sender:   cosmAddr.String(),
		Target:   common.HexToAddress("0x12345abcde").String(),
		Value:    math.NewIntFromBigInt(depositTxETH.Value()),
		GasLimit: big.NewInt(100_000).Bytes(),
		Data:     []byte{},
	})
```

File: e2e/stack_test.go [hunk 4]
```diff
@@ -324,13 +324,13 @@ func rollupFlow(t *testing.T, stack *e2e.StackConfig) {
...
-			math.NewIntFromBigInt(withdrawalTx.Value),
-			withdrawalTx.GasLimit,
-		),
+		testapp.ToTx(t, &rolluptypes.MsgInitiateWithdrawal{
+			Sender:   utils.EvmToCosmosAddress(*withdrawalTx.Sender).String(),
+				Target:   withdrawalTx.Target.String(),
+				Value:    math.NewIntFromBigInt(withdrawalTx.Value),
+				GasLimit: withdrawalTx.GasLimit.Bytes(),
+				Data:     []byte{},
+			}),
```
Comment (natebeauregard) [inline e2e/stack_test.go:333]: nit:
```suggestion
		testapp.ToTx(t, &rolluptypes.MsgInitiateWithdrawal{
			Sender:   utils.EvmToCosmosAddress(*withdrawalTx.Sender).String(),
			Target:   withdrawalTx.Target.String(),
			Value:    math.NewIntFromBigInt(withdrawalTx.Value),
			GasLimit: withdrawalTx.GasLimit.Bytes(),
			Data:     []byte{},
		}),
```

**Review by natebeauregard — APPROVED** [review]

**Pull request merged**